ui/core/player.ts

```diff
@@ -1299,4 +1299,30 @@ export class Player<SpecType extends Spec> {
 			}))
 		});
 	}
+
+	// Filter a matrix of spell IDs, min and max levels for a matching spell ID
+	getMatchingSpellActionId(src: number[][]): ActionId | null {
+		const match = src.find(([_, minLevel, maxLevel]) =>
+			(!minLevel || minLevel <= this.getLevel()) &&
+			(!maxLevel || maxLevel >= this.getLevel())
+		)
+
+		console.log(this.getLevel(), match)
+
+		if (match) return ActionId.fromSpellId(match[0])
+		return null
+	}
+
+	// Filter a matrix of item IDs, min and max levels for a matching item ID
+	getMatchingItemActionId(src: number[][]): ActionId | null {
+		const match = src.find(([_, minLevel, maxLevel]) =>
+			(!minLevel || minLevel <= this.getLevel()) &&
+			(!maxLevel || maxLevel >= this.getLevel())
+		)
+
+		console.log(this.getLevel(), match)
+
+		if (match) return ActionId.fromItemId(match[0])
+		return null
+	}
```

Comment:
These functions take in a matrix of up to 3-length arrays representing the item ID, a minimum level to display the buff, and a maximum level to display the buff (if applicable) and finds the correct entry corresponding to the player's current level and returns the ActionID from that entry.